github/resource_github_branch_test.go

```diff
@@ -0,0 +1,204 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-github/v29/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccGithubBranch_basic(t *testing.T) {
+
+	var (
+		reference github.Reference
+
+		name   = "basic"
+		repo   = "test-repo"
+		branch = "test-branch-" + acctest.RandString(5)
+		ref    = "refs/heads/" + branch
+		rn     = "github_branch." + name
+		id     = repo + ":" + branch
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGithubBranchDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckGithubBranchConfig(name, repo, branch),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubBranchExists(rn, id, &reference),
+					resource.TestCheckResourceAttr(rn, "repository", repo),
+					resource.TestCheckResourceAttr(rn, "branch", branch),
+					resource.TestCheckResourceAttr(rn, "source_branch", "master"),
+					resource.TestCheckResourceAttrSet(rn, "source_sha"),
+					resource.TestCheckResourceAttrSet(rn, "etag"),
+					resource.TestCheckResourceAttr(rn, "ref", ref),
+					resource.TestCheckResourceAttrSet(rn, "sha"),
+				),
+			},
+			{
+				Config: testAccCheckGithubBranchConfig(name, repo, branch),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubBranchExists(rn, id, &reference),
+					resource.TestCheckResourceAttr(rn, "repository", repo),
+					resource.TestCheckResourceAttr(rn, "branch", branch),
+					resource.TestCheckResourceAttr(rn, "source_branch", "master"),
+					resource.TestCheckResourceAttrSet(rn, "source_sha"),
+					resource.TestCheckResourceAttrSet(rn, "etag"),
+					resource.TestCheckResourceAttr(rn, "ref", ref),
+					resource.TestCheckResourceAttrSet(rn, "sha"),
+				),
+			},
+			{
+				ResourceName:      rn,
+				ImportState:       true,
+				ImportStateId:     fmt.Sprintf("%s:%s", repo, branch),
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
```

Comment:
The OCD in me says the previous for 4 values should be in line with the beginning of this one.

Reply:
Same, but you'll have to take that one up with `fmt`. This is out of my hands. Even your linter says so.

Reply:
😆 haha fine!